lib/libraries/c3/extensions/paginating-legend.js

```diff
@@ -0,0 +1,217 @@
+var d3 = require('d3');
+
+var isDateString = require('../../../utils/assert-date-string');
+
+module.exports = function(cols){
+  var self = this, chart, columns, domNode, pagination, range;
+
+  chart = this.view._artifacts.c3;
+  columns = [];
+  domNode = this.el().querySelector('.' + this.theme() + '-rendering');
+  range = Math.round((domNode.offsetHeight - 78) / 20);
+  pagination = this.view._artifacts.pagination = {
+    hidden: [],
+    labels: [],
+    position: 0,
+    range: range,
+    total: 0
+  };
+
+  // Reduce threshold for donut/pie charts
+  if (this.type() === 'donut' || this.type() === 'pie') {
+    pagination.range = pagination.range >= 5 ? 5 : pagination.range;
+  }
+
+  // Build a reversed column list
+  for (var i = 0; i < cols.length; i++) {
+    if (cols[i][0] !== 'x' && !isDateString(cols[i][1])) {
+      columns.push(cols[i][0]);
+    }
+  }
+
+  var legendEl = d3.select(domNode)
+    .append('svg')
+    .attr('class', 'keen-c3-legend')
+    .attr('width', 120)
+    .attr('height', domNode.offsetHeight - 10)
+    .style('right', '-120px')
```

Comment:
`-120px` is used a few times in this file to describe right margin: how about pulling it out to a variable?